### PR TITLE
fix!: use correct chain-id when filtering and checking txs

### LINKED ISSUE #2020: bank/tx: Can't send funds to account address from validator
### Summary of Bug

From an existing account that has funds, I can't send tia tokens to: 

- Existing celes accounts from hardspoons
- Newly freshly created accounts

### Version

v1.0.0-rc6

### Steps to Reproduce

```shell
consensus-validator-1-0:/$ celestia-appd tx bank send celestia1t5x0tdfepqjnez24keew4p46v89a0wn04ptz28 celestia1caqjp8w8udfeqldhvvlzwv8yd9xg7v8sdvhdav 100811405430utia --home /home/celestia --fees 21000000utia --keyring-backend test --gas 210000
auth_info:
  fee:
    amount:
    - amount: "21000000"
      denom: utia
    gas_limit: "210000"
    granter: ""
    payer: ""
  signer_infos: []
  tip: null
body:
  extension_options: []
  memo: ""
  messages:
  - '@type': /cosmos.bank.v1beta1.MsgSend
    amount:
    - amount: "100811405430"
      denom: utia
    from_address: celestia1t5x0tdfepqjnez24keew4p46v89a0wn04ptz28
    to_address: celestia1caqjp8w8udfeqldhvvlzwv8yd9xg7v8sdvhdav
  non_critical_extension_options: []
  timeout_height: "0"
signatures: []
confirm transaction before signing and broadcasting [y/N]: y
code: 0
codespace: ""
data: ""
events: []
gas_used: "0"
gas_wanted: "0"
height: "0"
info: ""
logs: []
raw_log: '[]'
timestamp: ""
tx: null
txhash: EF61BDB5C786FCD711293EE13E682872463FB8177470CE4CD0DF767CE7BF7482
```

Then query the tx 

```shell
consensus-validator-1-0:/$ celestia-appd query tx EF61BDB5C786FCD711293EE13E682872463FB8177470CE4CD0DF767CE7BF7482
Error: error in json rpc client, with http response metadata: (Status: 200 OK, Protocol HTTP/1.1). RPC error -32603 - Internal error: tx (EF61BDB5C786FCD711293EE13E682872463FB8177470CE4CD0DF767CE7BF7482) not found
``` 

### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

### PR DESCRIPTION
**Pull request by evan-forbes**
## Overview

consistently pass the correct chain-id to the application using `RequestPrepareProposal` and `RequestProcessProposal`

this PR still needs to update the go.mod to use the official tag of tendermint after its released

closes #2020 and #2021

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

**Comment by evan-forbes** [comment]
> unit tests failing?

yeah sorry, waiting for tests to finish locally but they should be fixed

File: test/util/testfactory/blob.go [hunk 1]
```diff
@@ -25,12 +25,12 @@ func GenerateRandomlySizedBlobs(count, maxBlobSize int) []types.Blob {
...
 	}
 
-	sort.Sort(types.BlobsByNamespace(blobs))
+	blobs = SortBlobs(blobs)
 	return blobs
 }
 
 // GenerateBlobsWithNamespace generates blobs with namespace ns.
-func GenerateBlobsWithNamespace(count int, blobSize int, ns appns.Namespace) types.BlobsByNamespace {
+func GenerateBlobsWithNamespace(count int, blobSize int, ns appns.Namespace) []types.Blob {
```
Comment (SweeXordious) [inline test/util/testfactory/blob.go:33]: [non-blocking]
why not keep using the `types.BlobsByNamespace` which already implements the `sort` interface

**Review by SweeXordious — DISMISSED** [review]
:shipit:

File: test/util/testfactory/blob.go (hunk 1 above)
Comment (evan-forbes) [reply test/util/testfactory/blob.go:33]: it was removed from core due to core not really being namespace aware, so yeah perhaps we should use that instead. atm I just tossed in a function

**Review by rootulp — APPROVED** [review]

**Review by SweeXordious — APPROVED** [review]

File: test/util/testfactory/blob.go [hunk 2]
```diff
@@ -78,3 +78,8 @@ func DelimLen(size uint64) int {
 	lenBuf := make([]byte, binary.MaxVarintLen64)
 	return binary.PutUvarint(lenBuf, size)
 }
+
+func SortBlobs(blobs []types.Blob) []types.Blob {
+	sort.Slice(blobs, func(i, j int) bool { return bytes.Compare(blobs[i].NamespaceID, blobs[j].NamespaceID) < 0 })
+	return blobs
+}
```
[the remark this answers is not in the document]
Comment (evan-forbes) [reply test/util/testfactory/blob.go:85]: good find, and yeah this is using the same as what we were using before (only for tests as well I believe)

**Pull request merged**